src/test/java/org/apache/sysds/test/functions/federated/monitoring/FederatedWorkerStatisticsTest.java

```diff
@@ -58,6 +87,54 @@ public void testWorkerStatisticsParsedCorrectly() {
 		Assert.assertNotEquals("Utilization stats parsed correctly", 0, model.utilization.size());
 	}
 
+	@Test
+	public void testWorkerStatisticsPerformance() throws InterruptedException {
```

Comment:
The reason for this is because the collection process happens in a different thread and in order to perform the experiments that I need to see the performance and scalability with different numbers of registered workers, I need to use only one method. I already have the data, we can remove it but the experiments would become irreproducible, would it be ok to add "Ignore" to the test and leave it there?

Reply:
First, if the test does not take that long to make, keep it.
But if it is running for 50+ seconds we need to reconsider.

Ideally we should have a way to test the multiple tests, to verify that it works with multiple worker threads underneath, 
that run in reasonable time.

I think in this case you know best, so if it is not possible at the moment/it takes to long for now we need to add a JIRA task for it.

Reply:
Only the execution definitely doesn't run for more than 50 seconds, BUT there is a sleep function in one of the performance tests which takes 70s so that the tool has enough time to gather 20 data entries. I left them there and put them with ignoring which, in my opinion, is the best solution since they don't really test any functionality but still can be used to check performance in case there are changes on the monitoring, let me know what you think.

Reply:
i can accept it, but it is unfortunate. and we would benefit from some smaller tests of this functionality.

Reply:
We do have functional tests in the same file, using only the workerService, which is the same part of the app being tested by the performance test.